9.0 port account bank statement import ofx

legalsylvain opened the pull request.

Commit:
Some changes for odoo 9.0

legalsylvain committed:
[REF] OCA convention;
[ADD] patch ofx parser to make this module work with some european bank like credit cooperatif;
[IMP] wizard view to display OFX implementation;

legalsylvain committed:
WIP

alexis-via commented:
Could you get the changes of this v8 PR https://github.com/OCA/bank-statement-import/pull/62

gurneyalex commented:
#44 also does a port of this module

legalsylvain commented:
@gurneyalex : yes my PR include https://github.com/OCA/bank-statement-import/pull/44 changes.

legalsylvain commented:
@alexis-via : I just rebased my work. #62 should be included now.

pedrobaeza commented:
Please check CIs status

ericzen commented:
So what's the OCA's position ? ofx import is valide for odoo 9 ? And in case of yes, how can i correct the error describes in #74 ?
For information i use 9.0_port_account_bank_statement_import_ofx branch.

pedrobaeza commented:
So you were referring to this PR? Any comment for something in progress should be reported here. Anyways, you don't describe any error. You should provide the error in the Odoo log and ideally the file you're trying to import to see the problem.

legalsylvain commented:
Hi @ericzen.

Thanks for testing. we are currently using this module without problem. So any log or detailled description of your problem are welcome ! 

regards.

ericzen commented:
unforgattable, there is no error in odoo's log (odoo-server.log). I've only this message box. That's why i give it to you.
I've test with demo.ofx files and same error.
Perhaps due to my install method.
Can you give me your method to install your addons ?
what i've done 
install python ofxparse
In settings/apps install account_bank_statement_import, account_bank_statement_import_multiline, account_bank_statement_import_ofx

ericzen commented:
Something can help me if you know what is Bank Reference. Indeed, perhaps my journal setting or Bank account setting is wrong.

ericzen commented:
OK It's work
Create a new folder and install account_bank_statement_import, account_bank_statement_import_ofx
work's fine now
Thanks

alexis-via reviewed (approved):
:+1:  when travis will be green. If we port to v10, we should start from here.

pedrobaeza commented:
@legalsylvain please check Travis status

alexis-via commented:
According to a very rapid test, this version of the ofx module also works on v10 without any modification (I tested it with my module account_bank_statement_import_usability available here for v10 https://github.com/akretion/odoo-usability/tree/10.0/account_bank_statement_import_usability that matches bank account when the OFX files has the national account number and Odoo has the IBAN)

NL66278 commented:
The tests fail because of this issue: https://github.com/OCA/bank-statement-import/issues/76 Must have something to do with the version of ofxparse I think. We might need to pin a certain version of this, but I do not know which would be the right one.

alexis-via commented:
I confirm this version works fine on v10 (tested with 2 different French banks, BNP and Banque populaires). I use ofxparse 0.15, which is the latest version cf https://pypi.python.org/pypi/ofxparse

pedrobaeza commented:
But please fix Travis

legalsylvain committed:
[FIX] travis / runbot error

pedrobaeza commented:
It continues the same...

alexis-via commented:
It is very strange, because with the latest version of ofxparse from pypi, I don't have this pb.

hbrunn commented on account_bank_statement_import_ofx/demo/demo_data.xml:
```diff
@@ -1,27 +1,23 @@
+        <field name="code">TBNKOFX</field>
+        <field name="type">bank</field>
+        <field name="sequence_id" ref="ofx_sequence"/>
+        <field name="user_id" ref="base.user_root"/>
```
this field doesn't exist any more

hbrunn reviewed (changes requested):
unfortunately, I can't help with the travis problem. A quick glance at https://github.com/jseutter/ofxparse/blob/master/ofxparse/ofxparse.py seems like the balance attribute is only set under certain conditions, any chance the test ofx file for the test simply doesn't contain whatever https://github.com/jseutter/ofxparse/blob/master/ofxparse/ofxparse.py#L910 wants?

alexis-via committed:
Skip lines with amount = 0 (blocker for this bank: www.banque-rhone-alpes.fr)

Use account number instead of bank number as 'name' (sync with code in v8)
Update import lines

alexis-via commented:
Now that ofxparse 0.15 has been release on pypi (cf https://pypi.python.org/pypi/ofxparse/0.15 released on June 30 2016, the date of your PR !), you can remove the file account_bank_statement_import_ofx/models/ofx.py
I confirm ofxparse 0.15 includes the fix
https://github.com/jseutter/ofxparse/commit/283f89c3246ed3fedccc3ef5c96078b7d5b94579

alexis-via commented:
BTW, I pushed a small PR to sylvain's branch: https://github.com/legalsylvain/bank-statement-import/pull/1 to update this PR (not to remove the OFX parser, but for other small things).

legalsylvain committed:
Merge pull request #1 from akretion/9sylvain-ofx-small-improvements

[9.0] OFX Skip lines with amount = 0 (blocker for this bank: www.banque-rhone-alpes.fr)

alexis-via commented:
I changed many more things in my PR of the port of the OFX module for v10. Maybe it will be a good backport them here once they are finished and stable.

legalsylvain commented:
Hi all, I'll merge PR of @alexis-via . I have no time to make the requested changes immediately. (removing obsolete ofx patch, ...) I'll review and accept new PR against this PR, or I'll do the changes in a while.
kind regards.

alexis-via commented on account_bank_statement_import_ofx/models/account_bank_statement_import.py:
```diff
@@ -38,15 +35,23 @@ def _parse_file(self, data_file):
                 banks = self.env['res.partner.bank'].search(
-                    [('owner_name', '=', transaction.payee)], limit=1)
+                    [('bank_name', '=', transaction.payee)], limit=1)
                 if banks:
```
This is wrong ; we should  keep the search on owner_name.

alexis-via commented:
@legalsylvain I think it will be easier to backport my v10 version to v9 when the v10 version will be finalized, as we continue to discover/fix issues in this PR during the review of the v10 PR https://github.com/OCA/bank-statement-import/pull/80. So let's continue the discussion on the v10 PR.

legalsylvain commented:
@alexis-via : Agree. Bette workflow. let keep it open this PR for the time being, to avoid other people working on porting this module in V9.0. In the meantime, let's go to review to V10.0 PR, and we'll backport your work.

Thanks for your work.

flotho commented:
Hi everyone,
As I can see the v10 migration has been merged.
Any chance to backport the job in this PR?
BTW, I tried this PR with the ofxparse from github master and still have the message :
```
    'balance_start': ofx.account.statement.balance,
AttributeError: 'Statement' object has no attribute 'balance'
```
Does the #80 solved this kind of issue?
Regards

alexis-via commented:
Yes, in #80 Travis is green, we don't have the error "balance_start': ofx.account.statement.balance,
AttributeError: 'Statement' object has no attribute 'balance'" any more. But we fixed this message by taking the last version of ofxparse from github... so it's strange that you still have it. And, in my case, I never had this message when using account_bank_statement_import_ofx in real life with real OFX files from several French banks using ofxparse 0.14 or 0.15... so for me it was a Travis-only issue.

flotho commented:
Hi @alexis-via , I personnaly tested the importation of 2 OFX from the same french bank group Credit Agricole. I have 2 files depending on different "federation". One of them is correctly imported the other one throw the famous "Statement' object has no attribute 'balance'"" error.
I've tested the importation with both ofxparse 0.15 from pypi and 0.14 from the github .
Is there any additionnal part of code that has changed /not been backported from v10 that could explain this behaviour?
Regards

HeO2SatDvr commented:
Hi Everyone.  I would like to contribute to this by testing with Canadian and US bank files but I don't know where to begin.  I have Odoo 9 community installed and functioning.  I have not updated the code for Odoo for about 9 months.  Do I need to have the latest Odoo code?  How and where to I install ofxparse?  How and where to I install the Odoo Addon developed here?

I am not a programmer.  I am a Management Consultant and I primarily work on Oracle systems.  I am exploring Odoo and thought testing might be a good way to contribute.

Thanks.

pedrobaeza commented:
Hi, @HeO2SatDvr, you can try the module using the guide in https://odoo-community.org/page/review, but unfortunately the problem here is a technical one I'm afraid.
